File: daq/slc/src/dqm/src/Histo1S.cc
#include "Histo1S.h"

#include "ShortArray.h"

using namespace Belle2;

Histo1S::Histo1S() throw() : Histo1()
{
  _data = new ShortArray(0);
}

Histo1S::Histo1S(const Histo1S& h) throw() : Histo1(h)
{
  _data = new ShortArray(h._data->size());
}

Histo1S::Histo1S(const std::string& name, const std::string& title, int nbinx, double min, double max)
throw() : Histo1(name, title, nbinx, min, max)
{
  _data = new ShortArray(nbinx + 2);
}

Histo1S::~Histo1S() throw() {}

std::string Histo1S::getDataType() const throw()
{
  return std::string("H1S");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/Histo1S.java
package b2daq.dqm.core;


public class Histo1S extends Histo1 {

	public Histo1S() {
		super();
		data = new ShortArray(0);
	}
	public Histo1S(String name, String title, int nbinx, double xmin, double xmax) {
		super(name, title, nbinx, xmin, xmax);
		data = new ShortArray(nbinx+2);
	}
	public Histo1S(Histo1S h) {
		super();
		data = new ShortArray(0);
	}
	public String getDataType() {
		return "H1S";
	}
	
}
